airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestination.java

```diff
@@ -0,0 +1,183 @@
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MongodbDestination extends BaseConnector implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MongodbDestination.class);
+
+  private static final String MONGO_URI = "mongodb://%s:%s@%s:%s/?authSource=%s&authMechanism=SCRAM-SHA-1";
+  private static final String HOST = "host";
+  private static final String PORT = "port";
+  private static final String DATABASE = "database";
+  private static final String AUTH_TYPE = "auth_type";
+  private static final String USERNAME = "username";
+  private static final String PASSWORD = "password";
+  private static final String AUTHORIZATION = "authorization";
+  private static final String LOGIN_AND_PASSWORD = "login/password";
+  private static final String AIRBYTE_DATA = "_airbyte_data";
+
+  private final MongoDBNameTransformer namingResolver;
+
+  public MongodbDestination() {
+    namingResolver = new MongoDBNameTransformer();
+  }
+
+  public static void main(String[] args) throws Exception {
+    final Destination destination = new MongodbDestination();
+    LOGGER.info("starting destination: {}", MongodbDestination.class);
+    new IntegrationRunner(destination).run(args);
+    LOGGER.info("completed destination: {}", MongodbDestination.class);
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try {
+      var uri = getMongoClientURI(config);
+      var client = getMongoClient(config.get(HOST).asText(), config.get(PORT).asInt(), uri);
+      var database = getMongoDatabase(client, config.get(DATABASE).asText());
+      return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED);
+    } catch (RuntimeException e) {
+      return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.FAILED)
+          .withMessage(e.getMessage() != null ? e.getMessage() : e.toString());
+    }
+  }
+
+  @Override
+  public AirbyteMessageConsumer getConsumer(JsonNode config,
+                                            ConfiguredAirbyteCatalog catalog,
+                                            Consumer<AirbyteMessage> outputRecordCollector) {
+    var uri = getMongoClientURI(config);
+    var client = getMongoClient(config.get(HOST).asText(), config.get(PORT).asInt(), uri);
+    var database = getMongoDatabase(client, config.get(DATABASE).asText());
+
+    Map<AirbyteStreamNameNamespacePair, MongodbWriteConfig> writeConfigs = new HashMap<>();
+    for (final ConfiguredAirbyteStream configStream : catalog.getStreams()) {
+      final AirbyteStream stream = configStream.getStream();
+      final String streamName = stream.getName();
+      final String collectionName = namingResolver.getRawTableName(streamName);
+      final String tmpCollectionName = namingResolver.getTmpTableName(streamName);
+
+      if (configStream.getDestinationSyncMode() == DestinationSyncMode.OVERWRITE) {
```

Comment:
I would write like  DestinationSyncMode.OVERWRITE ==  configStream.getDestinationSyncMode()  to prevent a NPE while calling getDestinationSyncMode(). I.e. "const.equals(arg)" rather than "arg.equals(const)". You never know if you receive arg actually defined

Reply:
Excellent spotted. Changed it.